api/controllers/user.controller.ts

```diff
@@ -1,56 +1,48 @@
 import { Request, Response } from 'express';
-import user from '../modules/user';
-
-const ERROR_MESSAGE_404 = 'User not found!';
+import UserService from '../secvices/user.service';
 
 class UserController {
-    get(req: Request, res: Response) {
+    public async get(req: Request, res: Response): Promise<void> {
         const { id } = req.params;
-        const result = user.get(id);
 
-        if (!result) {
-            return res.status(404).end(ERROR_MESSAGE_404);
-        }
+        const result = await UserService.getById(+id);
 
         res.json(result);
     }
 
-    add(req: Request, res: Response) {
+    public async add(req: Request, res: Response): Promise<void> {
         const { body } = req;
+        const result = await UserService.create(body);
 
-        res.json(user.add(body));
+        res.json(result);
     }
 
-    update(req: Request, res: Response) {
+    public async update(req: Request, res: Response): Promise<void> {
         const { id } = req.params;
         const { body } = req;
 
-        if (!user.exist(id)) {
-            return res.status(404).end(ERROR_MESSAGE_404);
-        }
+        const result = await UserService.update(+id, body);
 
-        res.json(user.update(id, body));
+        res.json(result);
     }
 
-    delete(req: Request, res: Response) {
+    public async delete(req: Request, res: Response): Promise<void> {
         const { id } = req.params;
 
-        if (!user.exist(id)) {
-            return res.status(404).end(ERROR_MESSAGE_404);
-        }
+        const result = await UserService.delete(+id);
 
-        res.json(user.delete(id));
+        res.json(result);
     }
 
-    list(req: Request, res: Response) {
-        const { login } = req.query;
-        const limit = Number(req.query.limit);
+    public async list(req: Request, res: Response): Promise<void> {
+        const {
+            login = '',
+            limit = 5
+        } = req.query;
 
-        if (isNaN(limit)) {
```

Comment:
Move this check to the service. There should not be any business logic in controller.

Reply:
This line marked as removed